tests/test_imgurfetcher.py

```diff
@@ -144,6 +265,145 @@ def test_get_image_from_album(self):
             mock_method.assert_called_once_with(self.album.id)
             self.assertTrue(result == self.gallery[-1])
 
+    """
+        tests that the constructor works properly
+    """
+    def test_constructor(self):
+
+        # test for wrong argument for settings file
+        with patch("bg_daemon.fetchers.imgurfetcher.os.path.join") as \
+                mock_method:
+
+            mock_method.return_value = None
+
+            with self.assertRaises(TypeError):
+                dummy_fetcher = imgurfetcher.imgurfetcher(None)
+
+        # test for corrupted json file
+        with patch("bg_daemon.fetchers.imgurfetcher.json.load") as mock_method:
+
+            # we write a json file that tries to overwrite the save method
+            corrupted_json = {"fetcher":{"query":None}}
+            mock_method.return_value = corrupted_json
+
+            with self.assertRaises(ValueError):
+                dummy_fetcher = imgurfetcher.imgurfetcher(self.settings_path)
+
+        # test for a "recent" mode fallback when initializing
+        with patch("bg_daemon.fetchers.imgurfetcher.json.load") as mock_method:
+
+            corrupted_json = {"fetcher":{"mode":"nonexistent"}}
+            mock_method.return_value = corrupted_json
+
+            dummy_fetcher = imgurfetcher.imgurfetcher(self.settings_path)
+            self.assertTrue(getattr(dummy_fetcher, "mode") is "recent")
```

Comment:
Are you testing equality or identity here and in the cases below? Just wanted to make sure that you didn't mean `==`.